notus/scanner/models/vulnerability.py

```diff
@@ -15,35 +15,46 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-from dataclasses import dataclass
-from typing import Dict
+from typing import Dict, List
 
-from ..models.advisory import Advisory
 from .packages.package import Package, PackageAdvisory
 
 
-@dataclass
-class PackageVulnerability:
+class Vulnerability:
     """Represents a vulnerable package on a host"""
 
-    host_ip: str
-    host_name: str
-    packages: Dict[Package, PackageAdvisory]
-    advisory: Advisory
-
-    def __init__(
-        self,
-        host_ip: str,
-        host_name: str,
-        package: Package,
-        fixed: PackageAdvisory,
-        advisory: Advisory,
-    ) -> None:
-        self.host_ip = host_ip
-        self.host_name = host_name
-        self.packages = {}
-        self.packages[package] = fixed
-        self.advisory = advisory
-
-    def add_package(self, package: Package, fixed: PackageAdvisory):
-        self.packages[package] = fixed
+    vulnerability: Dict[Package, List[PackageAdvisory]]
+
+    def __init__(self) -> None:
+        self.vulnerability = {}
+
+    def add_detect(self, package: Package, fixed: PackageAdvisory) -> None:
```

Comment:
why not just `add`? Through the class `Vulnerability` it is imho clear that adding it means it was detected contextually.